agents/node/vcxagent-core/src/agent/vcx-agent.js

```diff
@@ -0,0 +1,145 @@
+const { getLedgerAuthorAgreement, setActiveTxnAuthorAgreementMeta } = require('@absaoss/node-vcx-wrapper')
+const { createServiceLedgerCredDef } = require('./service-ledger-creddef')
+const { createServiceLedgerSchema } = require('./service-ledger-schema')
+const { createServiceVerifier } = require('./service-verifier')
+const { createServiceProver } = require('./service-prover')
+const { createServiceCredHolder } = require('./service-cred-holder')
+const { createServiceCredIssuer } = require('./service-cred-issuer')
+const { createServiceConnections } = require('./service-connections')
+const { provisionAgentInAgency } = require('../utils/vcx-workflows')
+const {
+  initVcxCore,
+  openVcxWallet,
+  openVcxPool,
+  vcxUpdateWebhookUrl,
+  shutdownVcx
+} = require('@absaoss/node-vcx-wrapper')
+const { createStorageService } = require('../state/storage-service')
+const { waitUntilAgencyIsReady } = require('../common')
+
+async function createVcxAgent ({ agentName, genesisPath, protocolType, agencyUrl, seed, usePostgresWallet, logger }) {
+  genesisPath = genesisPath || `${__dirname}/../../resources/docker.txn`
+
+  await waitUntilAgencyIsReady(agencyUrl, logger)
+
+  const storageService = await createStorageService(agentName)
+  if (!await storageService.agentProvisionExists()) {
+    const agentProvision = await provisionAgentInAgency(agentName, genesisPath, protocolType, agencyUrl, seed, usePostgresWallet, logger)
```

Comment:
I would prefer to use destructuring to pass that many arguments to avoid problems related to the ordering